src/server.rs

```diff
@@ -1423,6 +1423,59 @@ impl ServerRunner {
         });
         Error::verify_good(&status_code)
     }
+
+    /// Runs the server until it is cancelled.
+    ///
+    /// The server is shut down cleanly when `is_cancelled` returns true at which point the method
+    ///  returns.
+    ///
+    /// # Errors
+    ///
+    /// This fails when the server cannot be started.
```

Comment:
Thanks for taking care. I guess errors could happen anytime.

```suggestion
    /// This method could fail at any stage, i.e. while starting, running, or shutting down the server.
```